packages/flutter_tools/lib/src/macos/xcode.dart

```diff
@@ -198,6 +202,19 @@ class Xcode {
     final String appPath = _fileSystem.path.join(selectPath, 'Applications', 'Simulator.app');
     return _fileSystem.directory(appPath).existsSync() ? appPath : null;
   }
+
+  /// Gets the version number of the platform for the selected SDK.
+  Future<Version?> sdkPlatformVersion(EnvironmentType environmentType) async {
+    final RunResult runResult = await _processUtils.run(
+      <String>[...xcrunCommand(), '--sdk', getSDKNameForIOSEnvironmentType(environmentType), '--show-sdk-platform-version'],
+    );
+    if (runResult.exitCode != 0) {
+      _logger.printError('Could not find SDK Platform Version: ${runResult.stderr}');
+      return null;
+    }
+    final String versionString = runResult.stdout.trim();
+    return Version.parse(versionString);
```

Comment:
might parsing fail if there's extra STDOUT?

Reply:
Can you explain what you mean by extra STDOUT? Like if it has multiple lines?

Reply:
I wasn't quite sure what you meant, so I added some tests of potential things:
https://github.com/flutter/flutter/blob/62e0e9cbbed14a60be2f1f935fa1541db38089f7/packages/flutter_tools/test/general.shard/macos/xcode_test.dart#L295-L323

Reply:
I guess I was thinking of, in a future version of `Xcode`, the output of this looks like: `v1.2.3` and `Version.parse()` fails to parse it. I suppose in that case we'd return `null` here and we'd likely find out pretty quickly, so I think this is fine.